### PR TITLE
new matcher toBeNil

### PR DESCRIPTION
**Pull request by subinoy7**
adding new matcher ###toBeNil()

### Housekeeping

- [x] Unit tests

### CODE REVIEW

**Comment by tejasbubane** [comment]
@subinoy7 You don't need to close the PR. Can make changes and push on the same branch.

File: src/matchers/toBeNil/index.test.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+describe('.toBeNil', () => {
+  it('passes when given null', () => {
+    expect(null).toBeNil();
+  });
+
+  it('passes when undefined is given', () => {
+    expect(undefined).toBeNil();
+  });
+  it('fails when not undefined is given', () => {
```
Comment (tejasbubane) [inline src/matchers/toBeNil/index.test.js]: A better wording would be `fails when the value is not null or undefined`.

File: src/matchers/toBeNil/index.test.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  });
+});
+
+describe('.not.toBeNil', () => {
+  each([['true'], [{}], [true]]).it('passes when item is not null or undefined : %s', given => {
+    expect(given).not.toBeNil();
+  });
+
+  it('pass when undefined is given', () => {
+    expect(() => expect(null).not.toBeNil()).toThrowErrorMatchingSnapshot();
```
Comment (tejasbubane) [inline src/matchers/toBeNil/index.test.js]: test description says `undefined` but the value given is `null`

File: src/matchers/toBeNil/predicate.js [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export default received => {
+  return received == undefined || received == null;
```
Comment (tejasbubane) [inline src/matchers/toBeNil/predicate.js]: Comparing using triple equals `===` would be safer option.

Comment (tejasbubane) [reply src/matchers/toBeNil/predicate.js]: Also you can make it a single line function:

```
export default value => value === undefined || value === null;
```

File: src/matchers/toBeNil/predicate.test.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import each from 'jest-each';
+import predicate from './predicate';
+
+describe('toBeNil', () => {
+  it('returns true when given empty', () => {
+    expect(predicate(undefined)).toBe(true);
+  });
+
+  it('returns true when given an Array is empty', () => {
```
Comment (tejasbubane) [inline src/matchers/toBeNil/predicate.test.js]: Not an array.

File: src/matchers/toBeNil/predicate.test.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import each from 'jest-each';
+import predicate from './predicate';
+
+describe('toBeNil', () => {
+  it('returns true when given empty', () => {
```
Comment (tejasbubane) [inline src/matchers/toBeNil/predicate.test.js]: `undefined` instead of empty.

**Comment by subinoy7** [comment]
@tejasbubane thanks for highlighting all the points, I am going to fix this right now and will update it. Thanks a lot :smiley:

**Comment by tejasbubane** [comment]
> @tejasbubane thanks for highlighting all the points, I am going to fix this right now and will update it. Thanks a lot 😃

Cool 👍  Also don't forget to add yourself as a contributor.
Your name will appear at the bottom of the README !! 🎉 

To do that, run this command:

```console
yarn contributor <YOUR_GITHUB_USERNAME>
```

See the [contributing guide](https://github.com/mattphillips/jest-extended/blob/master/CONTRIBUTING.md) for more info.

**Comment by subinoy7** [comment]
@tejasbubane I have updated the files with changes.

File: README.md [hunk 6]
```diff
@@ -224,18 +223,6 @@ test('passes when given array values match any of the members in the set', () =>
...
-Use `.toSatisfyAll` when you want to use a custom matcher by supplying a predicate function that returns a `Boolean` for all values in an array.
-
-```js
-test('passes when all values in array pass given predicate', () => {
-  const isOdd = el => el % 2 === 1;
-  expect([1,3,5,7]).toSatisfyAll(isOdd);
-  expect([1,3,4,5,7]).not.toSatisfyAll(isOdd);
-});
-```
-
```
Comment (tejasbubane) [inline README.md]: Can you put this code back in?

File: src/matchers/toBeNil/index.test.js [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+    expect(() => expect('value').toBeNil()).toThrowErrorMatchingSnapshot();
+  });
+});
+
+describe('.not.toBeNil', () => {
+  each([['true'], [{}], [true]]).it('passes when value is not null or undefined : %s', given => {
+    expect(given).not.toBeNil();
+  });
+
+  it('passes when null is given', () => {
```
Comment (tejasbubane) [inline src/matchers/toBeNil/index.test.js]: `fails` when

**Comment by tejasbubane** [comment]
> @tejasbubane I have updated the files with changes.

Looks good 👍  I will let @mattphillips have a look and merge.

**Comment by mattphillips** [comment]
@subinoy7 great job all looks good to me! @tejasbubane thanks for taking the time to review this :smile:

**Pull request merged**